geth/process.py

```diff
@@ -293,13 +295,15 @@ def __init__(
         chain_name: str,
         base_dir: str | None = None,
         overrides: dict[str, Any] | None = None,
-        genesis_data: dict[str, Any] | None = None,
+        genesis_data: GenesisDataTypedDict | None = None,
     ):
         if overrides is None:
             overrides = {}
 
         if genesis_data is None:
-            genesis_data = GENESIS_JSON.copy()
+            genesis_data = GenesisDataTypedDict(**GENESIS_JSON)
```

Comment:
From my previous comment, this would become something like:

```suggestion
            genesis_data = GenesisData(GenesisDataTypedDict(**GENESIS_JSON))
```